### PR TITLE
OCPBUGS-22438: feat: add extra check for ready state on resiliency test

### PR DESCRIPTION
**Pull request by eggfoobar**

### CODE REVIEW

File: pkg/monitortestlibrary/pathologicaleventlibrary/duplicated_event_patterns.go [hunk 1]
```diff
@@ -909,3 +913,27 @@ func newTopologyAwareHintsDisabledDuringTaintTestsPathologicalEventMatcher(final
...
+		return eventInterval.Source == monitorapi.SourceAlert &&
+			strings.Contains(eventInterval.Message, "alert/TargetDown") &&
+			(eventInterval.StructuredLocator.Keys[monitorapi.LocatorNamespaceKey] == ocpAPINamespace ||
+				eventInterval.StructuredLocator.Keys[monitorapi.LocatorNamespaceKey] == ocpOAuthAPINamespace)
+	})
+	logrus.Infof("found %d OCP APIServer TargetDown intervals", len(ocpAPISeverTargetDownIntervals))
+	return &OverlapOtherIntervalsPathologicalEventMatcher{
+		delegate: &SimplePathologicalEventMatcher{
+			name:               "ConnectionErrorDuringSingleNodeAPIServerTargetDown",
+			messageReasonRegex: regexp.MustCompile(`dial tcp 172.30.0.1:443: connect: connection refused`),
```
Comment (stbenjam) [inline pkg/monitortestlibrary/pathologicaleventlibrary/duplicated_event_patterns.go:934]: Is that always guaranteed to be the IP? I imagine at some point dualstack or ipv6 single node might be planned and the API server URL will differ?

Comment (eggfoobar) [reply pkg/monitortestlibrary/pathologicaleventlibrary/duplicated_event_patterns.go]: Oh that's a good point, I updated the regex to account for either

**Pull request merged**